app/src/main/java/com/beemdevelopment/aegis/ui/ShareEntryActivity.java

```diff
@@ -0,0 +1,129 @@
+    private ImageView _qrImage;
+    private TextView _issuer;
+    private TextView _accountName;
+    private TextView _entriesCount;
+    private Button _nextButton;
+    private Button _previousButton;
+
+    private int _currentEntryCount = 1;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_share_entry);
+        _qrImage = findViewById(R.id.ivQrCode);
+        _issuer = findViewById(R.id.tvIssuer);
+        _accountName = findViewById(R.id.tvAccountName);
+        _entriesCount = findViewById(R.id.tvEntriesCount);
+        _nextButton = findViewById(R.id.btnNext);
+        _previousButton = findViewById(R.id.btnPrevious);
+
+        if (getSupportActionBar() != null){
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+            getSupportActionBar().setDisplayShowHomeEnabled(true);
+        }
+
+        Intent intent = getIntent();
+        _entries = (ArrayList<VaultEntry>) intent.getSerializableExtra("entries");
+
+        int controlVisibility = _entries.size() != 1 ? View.VISIBLE : View.INVISIBLE;
+        _entriesCount.setVisibility(controlVisibility);
+        _nextButton.setVisibility(controlVisibility);
+
+        _nextButton.setOnClickListener(v -> {
+            if (_currentEntryCount < _entries.size()) {
+                _previousButton.setVisibility(View.VISIBLE);
+                _currentEntryCount++;
+                generateQR();
+
+                if (_currentEntryCount == _entries.size()) {
+                    _nextButton.setText(R.string.done);
+                }
+            } else {
+                finish();
+            }
+        });
+
+        _previousButton.setOnClickListener(v -> {
+            if (_currentEntryCount > 1 ) {
+                _nextButton.setText(R.string.next);
+                _currentEntryCount--;
+                generateQR();
+
+                if (_currentEntryCount == 1) {
+                    _previousButton.setVisibility(View.INVISIBLE);
+                }
+            }
+        });
+
+        generateQR();
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                break;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+
+        return true;
+    }
+
+
+    private void generateQR() {
+        VaultEntry selectedEntry = _entries.get(_currentEntryCount - 1);
+        GoogleAuthInfo authInfo = new GoogleAuthInfo(selectedEntry.getInfo(), selectedEntry.getName(), selectedEntry.getIssuer());
+        QRCodeWriter writer = new QRCodeWriter();
+        BitMatrix bitMatrix = null;
+        try {
+            bitMatrix = writer.encode(authInfo.getUri().toString(), BarcodeFormat.QR_CODE, 512, 512);
+        } catch (WriterException e) {
+            e.printStackTrace();
```

Comment:
Should the execution of this method continue if encoding fails? If not, should the user be notified when this happens?